index.js

```diff
@@ -2,147 +2,87 @@
-    });
-  };
+module.exports = {
+  name: 'ember-component-css',
 
-  EmberApp.prototype._podStylePatterns = function() {
-    return this.registry.extensionsForType('css').map(function(extension) {
-      return new RegExp(extension + '$');
+  _getPodStyleFunnel() {
+    return new Funnel('app', {
+      srcDir: this._podDirectory(),
+      exclude: ['styles/**/*'],
+      include: this.allowedStyleExtentions.map(function(ext) { return new RegExp(ext + '$'); }),
+      allowEmpty: true,
+      annotation: 'Funnel (ember-component-css grab files)'
     });
-  };
+  },
 
-  EmberApp.prototype.styles = function() {
-    if (fs.existsSync('app/styles/' + this.name + '.css')) {
-      throw new SilentError('Style file cannot have the name of the application - ' + this.name);
+  _concatenatedPodStyles(wrappedStyles) {
+    var concatenatedStyles = [];
+
+    for (var i = 0; i < this.allowedStyleExtentions.length; i++) {
+      var extension = this.allowedStyleExtentions[i];
+      concatenatedStyles.push(new Concat(wrappedStyles, {
+        outputFile: 'pod-styles.' + extension,
+        inputFiles: ['**/*.' + extension],
+        sourceMapConfig: { enabled: true },
+        allowNone: true,
+        annotation: 'Concat (ember-component-css pod-styles.' + extension + ')'
+      }));
     }
 
-    var addonTrees = this.addonTreesFor('styles');
-    var external = this._processedExternalTree();
-    var styles = new Funnel(this.trees.styles, {
-      srcDir: '/',
-      destDir: '/app/styles'
-    });
-
-    var allTrees = [this.trees.app].concat(this.addonTreesFor('app'));
-    var podStyles = new Funnel(mergeTrees(allTrees, {overwrite: true}), {
-      include: this._podStylePatterns(),
-      exclude: [ /^styles/ ],
-      destDir: '/app',
-      description: 'Funnel: Pod Styles'
+    return new Merge(concatenatedStyles, {
+      annotation: 'Merge (ember-component-css merge concatenated styles)'
     });
+  },
 
-    var trees = [external].concat(addonTrees, podStyles, styles);
+  _allowedStyleExtentions() {
+    return this.registry.extensionsForType('css');
+  },
 
-    var stylesAndVendor = mergeTrees(trees, {
-      description: 'TreeMerger (stylesAndVendor)'
-    });
+  _podDirectory() {
+    return this.appConfig.podModulePrefix ? this.appConfig.podModulePrefix.replace(this.appConfig.modulePrefix, '') : '';
+  },
 
-    var options = { outputPaths: this.options.outputPaths.app.css };
-    options.registry = this.registry;
-    var preprocessedStyles = preprocessCss(stylesAndVendor, '/app/styles', '/assets', options);
+  included(app) {
+    this._super.included(app);
+    this.appConfig = app.project.config(); 
+    this.allowedStyleExtentions = this._allowedStyleExtentions();
+  },
 
-    var vendorStyles    = this.concatFiles(stylesAndVendor, {
-      inputFiles: this.vendorStaticStyles.concat(['vendor/addons.css']),
-      outputFile: this.options.outputPaths.vendor.css,
-      description: 'Concat: Vendor Styles'
+  treeForAddon(tree) {
```

Comment:
this is ES6 syntax, and not all nodes supported by ember-cli yet support this syntax.